n3fit/src/n3fit/msr.py

```diff
@@ -73,11 +75,19 @@ def msr_impose(nx=int(2e3), mode='All', scaler=None, photon=None):
     # 3.1 If a photon is given, compute the photon component of the MSR
     photon_c = 0.0
     if photon is not None:
-        photon_c = np.sum(
-            photon(
-                np.linspace(1e-5,1,100)[np.newaxis,:,np.newaxis]
-            )
-        )/100
+        
+        def make_lambert_grid(n_pts, x_min, x_max):
```

Comment:
Do we really need to copy? Can't you put this inside `vp`, and use the one from `eko`?

Reply:
Actually, I would prefer if the integral was done in the photon class/module etc. We also don't need anything fancy for this integral. Note that the MSR during the fit are computed by literally summing the arrays.

Reply:
If we are not too sensitive to the grid, better to use `np.logspace` and that's it. More code means more stuffs to maintain, and duplicated code looks always bad to me...

Reply:
Yes. I mean, I would just do:

```python
x, wgt = gen_integration_input(2e2) # or some other number of events
photon_array = photon(x)
photon_c = np.sum(photon_array * wgt)
```

`gen_integration_input` will give you a `logspace` at small x where you have more structure and a `linspace` at high x which is probably fine.

You can test it with `make_lambert_grid`. Maybe with that function you need a factor of 10 less events of course, but if the accuracy is similar I would do the simpler thing.

Reply:
I do not believe it will make so much the difference. Lambert grid was first introduced by @cschwan in PineAPPL, and it should smoothly interpolate from $log(x) \ \text{for}\ x \ll 1$ to $x \ \text{for}\ x \sim 1$, so I expect to more reasonable in the transition region, but not same results with $O(10)$ less points, most likely same order of magnitude.

Reply:
Then it is better my "no-code" solution. I would also expect not a big difference since I'm expecting the photon PDF to be quite smooth and easy to integrate.